open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/graph/LineageGraphQueryService.java

```diff
@@ -517,6 +539,50 @@ public LineageVertexResponse getEntityDetails(String guid) {
         return new LineageVertexResponse(lineageVertex);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public LineageTypesResponse getTypes() {
+        List<String> types = this.graphHelper.getResult(this::getTypes, this::handleTypesNotFoundException);
+        LineageTypesResponse typesResponse = new LineageTypesResponse();
+
+        if(CollectionUtils.isEmpty(types)) {
+            typesResponse.setRelatedHTTPCode(ERROR_TYPES_NOT_FOUND.getHTTPErrorCode());
+            typesResponse.setExceptionErrorMessage(ERROR_TYPES_NOT_FOUND.getFormattedErrorMessage());
+            typesResponse.setActionDescription(ERROR_TYPES_NOT_FOUND.getUserAction());
+            typesResponse.setExceptionUserAction(ERROR_TYPES_NOT_FOUND.getUserAction());
+            typesResponse.setExceptionErrorMessageId(ERROR_TYPES_NOT_FOUND.getErrorMessageId());
+            typesResponse.setExceptionClassName(OpenLineageException.class.getName());
+            return typesResponse;
+        }
+
+        typesResponse.setTypes(types);
+        return typesResponse;
+    }
+
+    private List<String> getTypes(GraphTraversalSource g) {
+       return g.V().not(hasLabel(ASSET_LINEAGE_VARIABLES, SUB_PROCESS)).label().dedup().toList();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public LineageNodeNamesResponse getNodes(NodeNamesSearchCriteria searchCriteria) {
+        List<String> nodeNames = this.graphHelper.getResult(this::getNodes, searchCriteria,
+                this::handleNodeNamesNotFoundException);
+        LineageNodeNamesResponse nodeNamesResponse = new LineageNodeNamesResponse();
+        nodeNamesResponse.setNames(nodeNames);
+        return nodeNamesResponse;
+    }
+
+    private List<String> getNodes(GraphTraversalSource g, NodeNamesSearchCriteria searchCriteria) {
+        return g.V().hasLabel(searchCriteria.getType()).has(DISPLAY_NAME_PROPERTY,
```

Comment:
instead of hasLabel() use has('vertex--label', 'type'), this uses the indexes created whilst hasLabel does not seem to use them